packages/flutter_tools/test/integration.shard/test_test.dart

```diff
@@ -198,6 +198,31 @@ void main() {
   testWithoutContext('flutter gold skips tests where the expectations are missing', () async {
     return _testFile('flutter_gold', automatedTestsDirectory, flutterTestDirectory, exitCode: isZero);
   });
+
+  testWithoutContext('flutter test should respect --serve-observatory', () async {
+    late final Process process;
+    late final StreamSubscription<String> sub;
+    try {
+      process = await _runFlutterTestConcurrent('trivial', automatedTestsDirectory, flutterTestDirectory,
+        extraArguments: const <String>['--start-paused', '--serve-observatory']);
+      final Completer<Uri> completer = Completer<Uri>();
+      final RegExp vmServiceUriRegExp = RegExp(r'((http)?:\/\/)[^\s]+');
+      sub = process.stdout.transform(utf8.decoder).listen((String e) {
```

Comment:
@bkonyi

Now I'm not a stream expert, but could it be non-deterministic how long the strings are that get emitted from the `process.stdout` stream? should we put a `LineSplitter` before the `.listen`?

Reply:
I'm not aware of any situations where the `stdout` stream has events that split the middle of a string (we use this same pattern all throughout the SDK test suites).

Reply:
As far as a process is concerned, it's just sending bytes to a file descriptor, though, right? not strings.

Reply:
Yeah, most likely. This is definitely a common pattern when scraping stdout of a child process.